chat/serializers.py

```diff
@@ -0,0 +1,169 @@
+import os
+import re
+from django.utils.timezone import now
+from rest_framework import serializers
+from chat.documents import Room, Message
+from core.settings.constants import FORBIDDEN_WORDS_SET
+
+MAX_PARTICIPANTS = int(os.getenv("MAX_PARTICIPANTS", 50))
+MIN_MESSAGE_LENGTH = int(os.getenv("MIN_MESSAGE_LENGTH", 1))
+MAX_MESSAGE_LENGTH = int(os.getenv("MAX_MESSAGE_LENGTH", 1000))
+
+
+class RoomSerializer(serializers.Serializer):
+    """
+    Serializer for chat rooms (conversations).
+
+    Fields:
+        - name (str): Unique identifier for the room (3-50 chars, letters, digits, dash, underscore).
+        - is_group (bool): Indicates if the room is a group chat (True) or private chat (False).
+        - participants (list[str]): List of user IDs participating in the room.
+
+    Validation:
+        - Room name must contain only allowed characters.
+        - Number of participants must not exceed MAX_PARTICIPANTS.
+        - Private rooms must have exactly 2 participants.
+    """
+
+    name = serializers.CharField(min_length=3, max_length=50)
+    is_group = serializers.BooleanField(default=False)
+    participants = serializers.ListField(
+        child=serializers.CharField(), allow_empty=False
+    )
+
+    def validate_name(self, value):
+        """
+        Ensure the room name only contains letters, digits, dash, or underscore.
+        """
+        if not re.match(r'^[a-zA-Z0-9_-]+$', value):
+            raise serializers.ValidationError(
+                "Room name can only contain letters, numbers, dash, underscore."
+            )
+        return value
+
+    def validate_participants(self, value):
+        """
+        Ensure participants list contains unique IDs and does not exceed the maximum limit.
+        """
+        unique_ids = list(set(value))
```

Comment:
Set conversion does not preserve order, which may cause confusing participant lists. Use OrderedDict to deduplicate while preserving order.
```
from collections import OrderedDict
unique_ids = list(OrderedDict.fromkeys(value))
```